src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/TimeoutsTest.java

```diff
@@ -170,16 +180,19 @@ public void cleanResources() {
     @Test
     public void testTimeoutsConfigured() throws Exception {
         final DatabaseEngine de = DatabaseFactory.getConnection(this.dbProps);
+        final Connection connection = de.getConnection();
 
-        int loginTimeoutInSeconds = DriverManager.getLoginTimeout();
-        int socketConnectionTimeoutInMs = de.getConnection().getNetworkTimeout();
-        int checkConnectionTimeoutInSec = de.getProperties().getCheckConnectionTimeout();
+        // Some engines (e.g. mysql) set the network timeout asynchronously.
+        await().atMost(Duration.ofSeconds(5))
+               .until(() ->
+                   TimeUnit.SECONDS.toMillis(SOCKET_TIMEOUT_SECONDS) == connection.getNetworkTimeout()
+               );
```

Comment:
This results in something like
`org.awaitility.core.ConditionTimeoutException: Condition with com.feedzai.commons.sql.abstraction.engine.impl.abs.TimeoutsTest was not fulfilled within 5 seconds.`

We should better identify this failure in the logs, and we can do the assertion right here:
```suggestion
        await("await network timeout set")
                .atMost(Duration.ofSeconds(5))
                .untilAsserted(() -> assertEquals("Is the socket timeout of the DB connection the expected?",
                        TimeUnit.SECONDS.toMillis(SOCKET_TIMEOUT_SECONDS), socketConnectionTimeoutInMs)
                );
```
This results in
```
org.awaitility.core.ConditionTimeoutException: Condition with alias 'await network timeout set' didn't complete within 5 seconds because assertion condition defined as a com.feedzai.commons.sql.abstraction.engine.impl.abs.TimeoutsTest Is the socket timeout of the DB connection the expected? expected:<20> but was:<0>.

........... stacktrace of the waiter .............

Caused by: java.lang.AssertionError: Is the socket timeout of the DB connection the expected? expected:<20> but was:<0>

........... stacktrace of the assert .............
```